internal/middleware/middleware_test.go

```diff
@@ -0,0 +1,586 @@
+					Status:  http.StatusInternalServerError,
+					Message: codes.GetErr(codes.ErrAssertClaims),
+					Data:    nil,
+				}
+				if !reflect.DeepEqual(&result, expected) {
+					t.Errorf("Want: %v, Got: %v", expected, result)
+				}
+			},
+		},
+		{
+			name: "Failure::ExtractUser:: user id not in claims",
+			setupFunc: func() (*http.Request, authentication.JWTService) {
+
+				req := httptest.NewRequest(http.MethodGet, "http://localhost:80", nil)
+				req.AddCookie(&http.Cookie{
+					Name:  "token",
+					Value: "123",
+				})
+				mockJwtSvc := mock.NewMockJWTService(mockCtrl)
+				token := jwtGo.Token{
+					Claims: jwtGo.MapClaims{},
+					Valid:  true,
+				}
+				mockJwtSvc.EXPECT().ValidateToken(gomock.Any()).Return(&token, nil)
+
+				return req, mockJwtSvc
+			},
+			validator: func(res *httptest.ResponseRecorder) {
+				if hit != false {
+					t.Errorf("Want: %v, Got: %v", false, hit)
+				}
+				by, _ := ioutil.ReadAll(res.Body)
+				result := model.Response{}
+				err := json.Unmarshal(by, &result)
+				if err != nil {
+					t.Log(err)
+					t.Fail()
+					return
+				}
+				expected := &model.Response{
+					Status:  http.StatusBadRequest,
+					Message: codes.GetErr(codes.ErrAssertUserid),
+					Data:    nil,
+				}
+				if !reflect.DeepEqual(&result, expected) {
+					t.Errorf("Want: %v, Got: %v", expected, result)
+				}
+			},
+		},
+		{
+			name: "Failure::ExtractUser:: user id not string",
+			setupFunc: func() (*http.Request, authentication.JWTService) {
+
+				req := httptest.NewRequest(http.MethodGet, "http://localhost:80", nil)
+				req.AddCookie(&http.Cookie{
+					Name:  "token",
+					Value: "123",
+				})
+				mockJwtSvc := mock.NewMockJWTService(mockCtrl)
+				token := jwtGo.Token{
+					Claims: jwtGo.MapClaims{"user_id": 1200},
+					Valid:  true,
+				}
+				mockJwtSvc.EXPECT().ValidateToken(gomock.Any()).Return(&token, nil)
+
+				return req, mockJwtSvc
+			},
+			validator: func(res *httptest.ResponseRecorder) {
+				if hit != false {
+					t.Errorf("Want: %v, Got: %v", false, hit)
+				}
+				by, _ := ioutil.ReadAll(res.Body)
+				result := model.Response{}
+				err := json.Unmarshal(by, &result)
+				if err != nil {
+					t.Log(err)
+					t.Fail()
+					return
+				}
+				expected := &model.Response{
+					Status:  http.StatusBadRequest,
+					Message: codes.GetErr(codes.ErrAssertUserid),
+					Data:    nil,
+				}
+				if !reflect.DeepEqual(&result, expected) {
+					t.Errorf("Want: %v, Got: %v", expected, result)
+				}
+			},
+		},
+	}
+
+	// to execute the tests in the table
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			// STEP 1: seting up all instances for the specific test case
+			res := httptest.NewRecorder()
+			req, jwt := tt.setupFunc()
+
+			// STEP 2: call the test function
+			middleware := NewTransactionMgmtMiddleware(&config.SvcConfig{
+				JwtSvc: config.JWTSvc{
+					JwtSvc: jwt,
+				},
+				Cfg: &config.Config{},
+			})
+			hit = false
```

Comment:
Instead of using global variables, use a local variable and pass this to the validator function. A pointer to this local variable can also be passed to the test handler function so that when the value is updated, the same variable will have this update.

Reply:
if I modify my existing handlerFunc to accept another arg as hit along with responseWriter and Request it is not accepting it as handler func

Reply:
done